recipe/meta.yaml

```diff
@@ -73,10 +80,14 @@ outputs:
         - mock
         - pytest
         - pip
+        - jax >= 0.3.24  # [osx or linux64]
+        - pytorch-cpu >=1.0  # [not win and not ppc64le]
+        - gym  # [py < 311]
       commands:
         - pip check
-        # TODO: figure out segfaults
-        # - pytest -v tests/
+        - MUJOCO_GL=osmesa pytest -v tests/  # [linux]
+        - pytest -v tests/  # [win]
+        - pytest -v tests/ -k 'not (test_no_error_warnings[env0] or test_no_error_warnings[env1] or test_play_relevant_keys or test_play_relevant_keys_with_env_attribute or test_video_size_no_zoom or test_video_size_zoom or test_keyboard_quit_event or test_pygame_quit_event or test_keyboard_relevant_keydown_event or test_keyboard_irrelevant_keydown_event or test_keyboard_keyup_event or test_play_loop_real_env or test_human_rendering) or test_play_relevant_keys_no_mapping'  # [osx]
```

Comment:
Let's please use a more readable setup for the test skips. I recommend the one used e.g. by arrow or [scipy](https://github.com/conda-forge/scipy-feedstock/blob/main/recipe/meta.yaml). Since `pytest -k` does substring matches, I removed redundant skips (e.g. skipping `test_play_relevant_keys` will automatically also skip `test_play_relevant_keys_with_env_attribute`). Alphabetically ordering the tests helps discover such instances.

```suggestion
        {% set tests_to_skip = "_not_a_real_test" %}
        # add a comment here like "the following tests are skipped [according to selector] because X"
        {% set tests_to_skip = tests_to_skip + " or test_human_rendering" %}                    # [osx]
        {% set tests_to_skip = tests_to_skip + " or test_keyboard_irrelevant_keydown_event" %}  # [osx]
        {% set tests_to_skip = tests_to_skip + " or test_keyboard_keyup_event" %}               # [osx]
        {% set tests_to_skip = tests_to_skip + " or test_keyboard_quit_event" %}                # [osx]
        {% set tests_to_skip = tests_to_skip + " or test_keyboard_relevant_keydown_event" %}    # [osx]
        {% set tests_to_skip = tests_to_skip + " or test_no_error_warnings[env0]" %}            # [osx]
        {% set tests_to_skip = tests_to_skip + " or test_no_error_warnings[env1]" %}            # [osx]
        {% set tests_to_skip = tests_to_skip + " or test_play_loop_real_env" %}                 # [osx]
        {% set tests_to_skip = tests_to_skip + " or test_play_relevant_keys" %}                 # [osx]
        {% set tests_to_skip = tests_to_skip + " or test_pygame_quit_event" %}                  # [osx]
        {% set tests_to_skip = tests_to_skip + " or test_video_size_no_zoom" %}                 # [osx]
        {% set tests_to_skip = tests_to_skip + " or test_video_size_zoom" %}                    # [osx]
        # need to specify opengl driver on linux (this comment is an example; feel free to improve)
        - export MUJOCO_GL="osmesa"  # [linux]
        - pytest -v tests/ -k 'not ({{ tests_to_skip }})'
```

Reply:
That looks indeed better! Thanks for the suggestion. However, skipping `test_play_relevant_keys` will also skip `test_play_relevant_keys_no_mapping`, which I explicitly added outside the parenthesis to put it back in, because it passes. I will modify your suggestion (somehow) and add it directly to line 104, without another variable, I think.

Reply:
In that case you can couple `test_play_relevant_keys` with something else that makes it unique in the serialization of the test name, e.g.
```
        {% set tests_to_skip = tests_to_skip + " or (test_play_relevant_keys and something_else)" %}  # [osx]
```
or if, every aspect (test path, test class, etc.) is the same as with `test_play_relevant_keys_no_mapping`, then the following should work:
```
        # avoid skipping a passing test due to on identical substrings
        {% set tests_to_skip = tests_to_skip + " or (test_play_relevant_keys and not test_play_relevant_keys_no_mapping)" %}  # [osx]
```

Reply:
Yes, the latter is true. I was a bit disappointed that in pytest there is no option to specify the full test name.

Your suggestion

```python
{% set tests_to_skip = tests_to_skip + " or (test_play_relevant_keys and not test_play_relevant_keys_no_mapping)" %}  # [osx]
```

looks better than what I was about to be doing:

```python
pytest -v tests/ -k 'not ({{ tests_to_skip }}) or test_play_relevant_keys_no_mapping'
```

because `test_play_relevant_keys_no_mapping` is closer to `test_play_relevant_keys`.

Reply:
> Yes, the latter is true. I was a bit disappointed that in pytest there is no option to specify the full test name.

Well, you can specify the full test name (you just have to use it as pytest would serialize it), but if there are other tests that are named exactly the same just by adding some extra characters to the test name, then it would be a huge jump in complexity of pytest's implementation to not capture those as well (and in many situation, this behaviour is desired - so a feature, not a bug).

I'll note though that perfect is definitely the enemy of the good here. I care about enabling the test suite _at all_, not that we'd be skipping one passing test too many on one platform. I'd be more than happy to just use
```
{% set tests_to_skip = tests_to_skip + " or test_play_relevant_keys" %}  # [osx]
```
In fact I prefer it because the extra noise in the recipe is not worth unskipping a test IMO.

If you really want to fix it, the easiest option would be to avoid identical substrings in tests if they do very different thing (ideally upstream; would be possible through a patch, but for that as well, the complexity vs. benefit trade-off is negative).

Reply:
> Is there a way in github I can modify your suggested changes?

You could if you become maintainer on this feedstock (then you could edit the suggestion in my comment & commit it) - would you be interested in that (given your apparent interest in gymnasium & great work in this PR)? 🙃 

> I really just want to change your sample comments. Or should I commit your suggestion, amend changes and force push?

Force-pushing is fine, especially if it helps future git archeology. I tend to clean up the commits in my PRs before merging them so that future me will have an easier time understanding why something was done when I have to dig into the git history.

But in this case, you don't even have to do that. You can just copy & paste my suggestion into your checkout locally & commit it. I don't need the attribution that GH generates. That's not to say that there can't be cases where such metadata operations are desirable (and they can of course be replicated also locally using `git commit --author=` and/or `Co-Authored-By: Name <email>` in the commit message).

Reply:
> You could if you become maintainer on this feedstock (then you could edit the suggestion in my comment & commit it) - would you be interested in that (given your apparent interest in gymnasium & great work in this PR)?

Thanks a lot for the offer and your kind words, but probably I could not handle the time it takes. My main motivation for this was actually that I need to maintain some installations and installation instructions in a university to install gymnasium-atari and I like installations with mamba much more than with pip. I saw #16 and that shimmy-atari together with the failing tests are the main blocker. Maybe I can also help to upgrade the package version, since I know that some of my patches have to be adapted for that. I might try to bring autorom and pettingzoo also to conda-forge next, if I can find the time for it. So, its not that I am not interested in it, but I cannot make any promises in the long run.

Reply:
> So, its not that I am not interested in it, but I cannot make any promises in the long run.

Maintainership is not a promise to commit time, we're all volunteers here, and no-one can force use to do this unpaid work. But it does give you easier handling and more of a say. Anyway, it's an open invitation. There's never a shortage of things to do, as you discovered. ;-)